polaris-core/src/main/java/org/apache/polaris/core/persistence/impl/PolarisResolutionManifestBuilder.java

```diff
@@ -0,0 +1,191 @@
+    addTopLevelName(PolarisEntityConstants.getRootContainerName(), PolarisEntityType.ROOT, true);
+  }
+
+  @Override
+  public ResolutionManifestBuilder withRootContainerEntity(
+      ResolvedPolarisEntity rootContainerEntity) {
+    this.rootContainerEntity = rootContainerEntity;
+    return this;
+  }
+
+  @Override
+  public ResolutionManifestBuilder addTopLevelName(
+      String entityName, PolarisEntityType entityType, boolean optional) {
+    addedTopLevelNames.put(entityName, entityType);
+    if (optional) {
+      primaryResolver.addOptionalEntityByName(entityType, entityName);
+    } else {
+      primaryResolver.addEntityByName(entityType, entityName);
+    }
+    return this;
+  }
+
+  @Override
+  public ResolutionManifestBuilder addPath(ResolverPath resolverPath, String catalogRoleName) {
+    return addPathInternal(resolverPath, catalogRoleName);
+  }
+
+  @Override
+  public ResolutionManifestBuilder addPath(ResolverPath resolverPath, Namespace namespace) {
+    return addPathInternal(resolverPath, namespace);
+  }
+
+  @Override
+  public ResolutionManifestBuilder addPath(
+      ResolverPath resolverPath, TableIdentifier tableIdentifier) {
+    return addPathInternal(resolverPath, tableIdentifier);
+  }
+
+  private ResolutionManifestBuilder addPathInternal(ResolverPath path, Object key) {
+    primaryResolver.addPath(path);
+    pathLookup.put(key, currentPathIndex);
+    addedPaths.add(path);
+    ++currentPathIndex;
+    return this;
+  }
+
+  @Override
+  public ResolutionManifestBuilder addPassthroughPath(
+      ResolverPath resolverPath, TableIdentifier tableIdentifier) {
+    return addPassthroughPathInternal(resolverPath, tableIdentifier);
+  }
+
+  @Override
+  public ResolutionManifestBuilder addPassthroughPath(
+      ResolverPath resolverPath, Namespace namespace) {
+    return addPassthroughPathInternal(resolverPath, namespace);
+  }
+
+  private ResolutionManifestBuilder addPassthroughPathInternal(
+      ResolverPath resolverPath, Object key) {
+    addPathInternal(resolverPath, key);
+    passthroughPaths.put(key, resolverPath);
+    return this;
+  }
+
+  @Override
+  public ResolutionManifestBuilder notFoundExceptionMapper(
+      Function<EntityNotFoundException, RuntimeException> notFoundExceptionMapper) {
+    this.notFoundExceptionMapper = notFoundExceptionMapper;
+    return this;
+  }
+
+  @Override
+  public ResolutionManifest buildResolved() {
+    return buildResolved(null);
+  }
+
+  @Override
+  public ResolutionManifest buildResolved(PolarisEntitySubType subType) {
+
+    try {
+      var resolver = primaryResolver.buildResolved();
+
+      List<PrincipalRoleEntity> activatedPrincipalRoles =
+          resolver.getResolvedCallerPrincipalRoles().stream()
+              .map(ce -> PrincipalRoleEntity.of(ce.getEntity()))
+              .collect(Collectors.toList());
+      this.authenticatedPrincipal.setActivatedPrincipalRoles(activatedPrincipalRoles);
```

Comment:
Why is this pulled out of the `Resolver` process? I mean, I don't think it should be in there, but I think it should be part of the construction of the `AuthenticatedPolarisPrincipal`, not here. I'd scope this change to keep the logic inside the `Resolver` process for now, then extract the PrincipalRole resolution as part of a new change.